front/pages/w/[wId]/builder/labs/transcripts/index.tsx

```diff
@@ -0,0 +1,353 @@
+  const [isLoading, setIsLoading] = useState(false);
+  const [isGDriveConnected, setIsGDriveConnected] = useState(false);
+  const [assistantSelected, setAssistantSelected] =
+    useState<LightAgentConfigurationType | null>(null);
+  const sendNotification = useContext(SendNotificationsContext);
+  const [emailToNotify, setEmailToNotify] = useState<string | null>("");
+  const [agentsFetched, setAgentsFetched] = useState(false);
+  const [isActive, setIsActive] = useState(false);
+
+  const { agentConfigurations } = useAgentConfigurations({
+    workspaceId: owner.sId,
+    agentsGetView: "list",
+    sort: "priority",
+  });
+
+  const agents = agentConfigurations.filter((a) => a.status === "active");
+
+  const makePatchRequest = async (data: any, successMessage: string) => {
+    await fetch(`/api/w/${owner.sId}/labs/transcripts`, {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(data),
+    }).then((response) => {
+      if (!response.ok) {
+        throw new Error("Failed to update");
+      }
+      sendNotification({
+        type: "success",
+        title: "Success!",
+        description: successMessage,
+      });
+    });
+  };
+
+  const updateAssistant = async (assistant: LightAgentConfigurationType) => {
+    setAssistantSelected(assistant);
+    const data = {
+      agentConfigurationId: assistant.sId,
+      provider,
+    };
+    const successMessage =
+      "The assistant that will help you summarize your transcripts has been set to @" +
+      assistant.name;
+    await makePatchRequest(data, successMessage);
+  };
+
+  const updateEmailToNotify = async (email: string) => {
+    setEmailToNotify(email);
+    const data = {
+      email: email,
+      agentConfigurationId: assistantSelected?.sId,
+      provider,
+    };
+    const successMessage = "The email to notify has been set to " + email;
+    await makePatchRequest(data, successMessage);
+  };
+
+  const updateIsActive = async (isActive: boolean) => {
+    setIsActive(isActive);
+    const data = {
+      isActive,
+      agentConfigurationId: assistantSelected?.sId,
+      provider,
+    };
+    const successMessage = isActive
+      ? "We will start summarizing your transcripts."
+      : "We will no longer summarize your transcripts.";
+    await makePatchRequest(data, successMessage);
+  };
+
+  const handleSelectAssistant = async (
+    assistant: LightAgentConfigurationType
+  ) => {
+    return updateAssistant(assistant);
+  };
+
+  const handleSaveEmailToNotify = async () => {
+    if (emailToNotify) {
+      return updateEmailToNotify(emailToNotify);
+    }
+  };
+
+  const handleSetIsActive = async (isActive: boolean) => {
+    return updateIsActive(isActive);
+  };
+
+  useEffect(() => {
+    if (agentConfigurations.length > 0) {
+      setAgentsFetched(true);
+    }
+  }, [agentConfigurations]);
+
+  useEffect(() => {
```

Comment:
If you use a hook here (e.g. `useAgentConfiguration`) you could remove a lot of boilerplate + complexity from here.